x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/90_behavioral_analytics_event_post.yml

```diff
@@ -0,0 +1,163 @@
+          search:
+            query: "test-query"
+---
+"Post Search Analytics Event – missing search query":
+  - do:
+      catch: "bad_request"
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "search"
+        body:
+          session:
+            id: "123"
+          user:
+            id: "456"
+  - match: { error.type: "illegal_argument_exception" }
+---
+"Post SearchClick Analytics Event":
+  - do:
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "search_click"
+        body:
+          session:
+            id: "123"
+          user:
+            id: "456"
+          page:
+            url: "https://www.elastic.co"
+          search:
+            query: "test-query"
+---
+"Post SearchClick Analytics Event – missing search query":
+  - do:
+      catch: "bad_request"
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "search_click"
+        body:
+          session:
+            id: "123"
+          user:
+            id: "456"
+          page:
+            url: "https://www.elastic.co"
+  - match: { error.type: "illegal_argument_exception" }
+---
+"Post SearchClick Analytics Event – missing page url":
+  - do:
+      catch: "bad_request"
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "search_click"
+        body:
+          session:
+            id: "123"
+          user:
+            id: "456"
+          search:
+            query: "test-query"
+  - match: { error.type: "illegal_argument_exception" }
+---
+"Post Analytics Event - analytics collection does not exist":
+  - do:
+      catch: "missing"
+      behavioral_analytics.post_event:
+        collectionName: test-nonexistent-analytics-collection
+        eventType: "pageview"
+        body:
+          session:
+            id: "123"
+          user:
+            id: "456"
+          page:
+            url: "https://www.elastic.co"
+---
+"Post Analytics Event - event type does not exist":
+  - do:
+      catch: "bad_request"
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "nonexistent-event-type"
+        body:
+          session:
+            id: "123"
+          user:
+            id: "456"
+          page:
+            url: "https://www.elastic.co"
+  - match: { error.type: "action_request_validation_exception" }
+---
+"Post Analytics Event - missing session id":
+  - do:
+      catch: "bad_request"
+      behavioral_analytics.post_event:
+        collectionName: my-test-analytics-collection
+        eventType: "pageview"
+        body:
+          user:
+            id: "456"
+          page:
+            url: "https://www.elastic.co"
+  - match: { error.type: "illegal_argument_exception" }
+---
+"Post Analytics Event - missing user id":
```

Comment:
Will all pageViews have user ID? Isn't it possible that a user hasn't logged in yet?

Reply:
The JS tracker add a user id to each query.
This is a cookie that is longer than session id (24h by default vs 30 min) to allow us to identify unique visitor within a day.

Reply:
Got it, thanks for the explanation! I get that application-level user id is sent as part of other metadata then. Thanks!